merkledag.go

```diff
@@ -7,7 +7,39 @@ import (
 	cid "github.com/ipfs/go-cid"
 )
 
-var ErrNotFound = fmt.Errorf("merkledag: not found")
+// ErrNotFound is used to signal when a Node could not be found. The specific
+// meaning will depend on the DAGService implementation, which may be trying
+// to read nodes locally but also, trying to find them remotely.
+//
+// The Cid field can be filled in to provide additional context.
+type ErrNotFound struct {
+	Cid cid.Cid
+}
```

Comment:
Exposed Cid thinking that this might be used from outside this module.

It is ugly though that you have to instantiate the struct whenever you want to check if `errors.Is(whatever, ErrNotFound{})`.

Reply:
We _can_ add an `IsNotFound` function.

Reply:
Added, and made it equivalent to `Is`.